pkg/registry/core/service/ipallocator/ipallocator.go

```diff
@@ -343,11 +348,16 @@ func (a *Allocator) allocateFromRange(iterator func() netip.Addr, svc *api.Servi
 		}
 		// address is not present on the cache, try to allocate it
 		err = a.createIPAddress(name, svc, "dynamic")
-		// an error can happen if there is a race and our informer was not updated
-		// swallow the error and try with the next IP address
+	// an error can happen if there is a race and our informer was not updated
+		// in that case, swallow the error and try with the next IP address,
+		// but abort on other errors. Ref: https://issues.k8s.io/135333
 		if err != nil {
+			if apierrors.IsAlreadyExists(err) {
```

Comment:
The `createIPAddress` function return a `ErrAllocated` error when the kube-apiserver returns an AlreadyExists error. 

https://github.com/aojea/kubernetes/blob/63a664a27e6d94e896a448e5e05e87bff8372db8/pkg/registry/core/service/ipallocator/ipallocator.go#L159

Reply:
uppps

Reply:
updated and added unit test, great catch